app/forms/residential_address_form.rb

```diff
@@ -1,12 +1,18 @@
 class ResidentialAddressForm < Form
-  set_attributes_for :household, :residential_street, :residential_street_2, :residential_city, :residential_zip_code, :has_mailing_address
-  validates_presence_of :residential_street, message: 'Please fill in the street address.'
-  validates :residential_street_2, length: { maximum: 128, too_long: 'Please enter a shorter unit or apartment.' }
-  validates_presence_of :residential_city, message: 'Please fill in the city.'
-  validates_presence_of :residential_zip_code, message: 'Please fill in the ZIP code.'
-  validates_presence_of :has_mailing_address, message: 'Please choose where to send mail.'
+  set_attributes_for :household, :residential_street, :residential_street_2, :residential_city, :residential_zip_code, :has_mailing_address, :registered_homeless
+  validates_presence_of :residential_street, message: 'Please fill in the street address.', if: :registered_homeless_unselected
+  validates :residential_street_2, length: { maximum: 128, too_long: 'Please enter a shorter unit or apartment.' }, if: :registered_homeless_unselected
+  validates_presence_of :residential_city, message: 'Please fill in the city.', if: :registered_homeless_unselected
+  validates_presence_of :residential_zip_code, message: 'Please fill in the ZIP code.', if: :registered_homeless_unselected
+  validates_presence_of :has_mailing_address, message: 'Please choose where to send mail.', if: :registered_homeless_unselected
 
   def save
     household.update(attributes_for(:household))
   end
+
+  private
+
+  def registered_homeless_unselected
+    registered_homeless.blank?
```

Comment:
Just checking that this is the right method, not `registered_homeless_unfilled`?

Reply:
`registered_homeless_unfilled` doesn't work since it's reading off the household which hasn't been saved yet...

But now I realize that having it as an enum means that this shouldn't work either? Somehow this passed the test at the time but I changed it to `registered_homeless == 'unfilled'`